### PR TITLE
Initial implementation of protobuf-delta-lake-sink

### PR DESCRIPTION
**Pull request by ChewingGlass**

### CODE REVIEW

**Comment by ke6jjj** [comment]
This isn't production code yet, so it seems silly to require reviews at this stage. So... approved!

**Review by ke6jjj — APPROVED** [review]

File: protobuf-delta-lake-sink/src/proto/parse.rs [hunk 1]
```diff
@@ -0,0 +1,560 @@
...
+use protobuf::reflect::{EnumDescriptor, ReflectValueBox};
+use protobuf::{
+    reflect::{FieldDescriptor, MessageDescriptor, ReflectValueRef, RuntimeType},
+    MessageDyn,
+};
+
+use super::get_delta_schema;
+
+trait ReflectBuilder: ArrayBuilder {
+    fn append_value(&mut self, v: Option<ReflectValueRef>) -> ();
```
Comment (maplant) [inline protobuf-delta-lake-sink/src/proto/parse.rs]:
```suggestion
    fn append_value(&mut self, v: Option<ReflectValueRef>);
```

functions returning nil is the default, you don't need to specify it

File: protobuf-delta-lake-sink/src/proto/parse.rs [hunk 2]
```diff
@@ -0,0 +1,560 @@
...
+            .len(length)
+            .child_data(child_data)
+            .null_bit_buffer(None);
+
+        let array_data = unsafe { builder.build_unchecked() };
+        Arc::new(StructArray::from(array_data))
+    }
+
+    fn finish_cloned(&self) -> ArrayRef {
+        panic!("Not implemented")
```
Comment (maplant) [inline protobuf-delta-lake-sink/src/proto/parse.rs]:
```suggestion
        unimplemented!()
```
or implement this

File: protobuf-delta-lake-sink/src/main.rs [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+        Err(DeltaTableError::NotATable(_)) => {
+            println!("It doesn't look like our delta table has been created");
+            DeltaOps(table_raw.into())
+                .create()
+                .with_columns(delta_schema)
+                .with_partition_columns(vec!["date"])
+                .await
+                .unwrap()
+        }
+        Err(err) => Err(err).unwrap(),
```
Comment (maplant) [inline protobuf-delta-lake-sink/src/main.rs]: Is this to panic on the error? tokio main functions are allowed to return `Result<()>`

Comment (ChewingGlass) [reply protobuf-delta-lake-sink/src/main.rs]: Just had it return result and used anyhow to coerce everything. Anyhow is super nice

**Pull request merged**